tests/opentelemetry/test_context.py

```diff
@@ -129,3 +133,38 @@ async def coro(i):
         await coro(2)
         await coro(3)
         await coro(4)
+
+
+def test_otel_baggage(oteltracer):
+    with oteltracer.start_as_current_span("otel-baggage-inject") as span:  # noqa: F841
+        baggage_context = set_baggage("key1", "value1")
+        baggage_context = set_baggage("key2", "value2", baggage_context)
+        attach(baggage_context)
+        with ddtrace.tracer.trace("ddtrace-baggage-inject") as ddspan:
+            assert ddspan.context.get_baggage_item("key1") == "value1"
+            assert ddspan.context.get_baggage_item("key2") == "value2"
+
+
+def test_dd_baggage(oteltracer):
+    with ddtrace.tracer.trace("ddtrace-baggage") as ddspan:
+        ddspan.context.set_baggage_item("key1", "value1")
+        ddspan.context.set_baggage_item("key2", "value2")
+        assert get_baggage("key1") == "value1"
+        assert get_baggage("key2") == "value2"
+
```

Comment:
Can we add a test for an intermixed DD + OTEL set/get to make sure the last ones win

```suggestion

def test_mixed_otel_dd_baggage(oteltracer):
    with ddtrace.tracer.trace("ddtrace-baggage") as ddspan:
        ddspan.context.set_baggage_item("key1", "dd1")
        attach(set_baggage("key1", "otel1"))
        attach(set_baggage("key2", "otel2"))
        ddspan.context.set_baggage_item("key2", "dd2")
        assert get_baggage("key1") == "otel1"
        assert get_baggage("key2") == "dd2"

```